docs/source/config_file.rst

```diff
@@ -65,234 +65,346 @@ unfortunate, and is subject to change in future versions.
 
-- ``pdb`` (Boolean, default False) invokes pdb on fatal error.
+4.  Selectively disallow untyped function definitions only within the ``mycode.foo``
+    package -- that is, only for function definitions defined in the
+    ``mycode/foo`` directory.
+    
+5.  Selectively *disable* the "function is returning any" warnings within only
+    ``mycode.bar``. This overrides the global default.
 
-- ``show_traceback`` (Boolean, default False) shows traceback on fatal
-  error.
+6.  Suppress any error messages generated when your codebase tries importing the module 
+    ``somelibrary``. (We assume here that ``somelibrary`` is some 3rd party
+    library missing type hints).
 
-- ``dump_type_stats`` (Boolean, default False) dumps stats about type
-  definitions.
 
-- ``dump_inference_stats`` (Boolean, default False) dumps stats about
-  type inference.
+.. _per-module-flags:
 
-- ``incremental`` (Boolean, default True) enables :ref:`incremental
-  mode <incremental>`.
+Per-module and global options
+*****************************
 
-- ``cache_dir`` (string, default ``.mypy_cache``) stores module cache
-  info in the given folder in :ref:`incremental mode <incremental>`.
-  The cache is only read in incremental mode, but it is always written
-  unless the value is set to ``/dev/null`` (UNIX) or ``nul``
-  (Windows).
+The following config options may be set either globally (in the ``[mypy]`` section)
+or on a per-module basis (in sections like ``[mypy-foo.bar]``).
 
-- ``quick_and_dirty`` (Boolean, default False) enables :ref:`quick
-  mode <quick-mode>`.
+If you set an option both globally and for a specific module, the module configuration
+options take precedence. This lets you set global defaults and override them on a
+module-by-module basis. If multiple pattern sections match a module, the options from the
+most specific section are used where they disagree. See above for more details.
 
-- ``show_error_context`` (Boolean, default False) shows
-  context notes before errors.
+.. _config-file-import-discovery-per-module:
 
-- ``show_column_numbers`` (Boolean, default False) shows column numbers in
-  error messages.
+Import discovery
+----------------
 
+For more information, see the :ref:`import discovery <import-discovery>`
+section of the command line docs.
 
-.. _per-module-flags:
+Note: this section describes options that can be used both globally and per-module.
+See below for a list of import discovery options that may be used
+:ref:`only globally <config-file-import-discovery-global>`.
 
-Per-module flags
-****************
+``ignore_missing_imports`` (boolean, default False) 
+    Suppress error messages about imports that cannot be resolved.
 
-The following flags may vary per module.  They may also be specified in
-the global section; the global section provides defaults which are
-overridden by the pattern sections matching the module name.
+    Note that if pattern matching is used, the pattern should match the
```

Comment:
The term "pattern matching" seems to be used only here and for the next option. Maybe instead use a phrase like "if this option is used in a per-module section, the module name pattern should match the [etc.]"?

Reply:
Done!